### ISSUE TITLE
test with batch size 1

### ISSUE THREAD

**Issue by pengxj** [JiaweiShiCV/Amend-Representation-Module#17]
There is a problem. In testing, one can not use a batch representation for a certain image. Especially, on RAF-DB, you are doing batch recognition (all images belong to the same category in most cases) if shuffe=False.
So, the accuracy is convinced only bs = 1.

**Comment by pengxj** [comment]
And I think if shuffe=True in testing, the performance will degrade seriously.

**Comment by msy1412** [comment]
I have tried the Pre-trained model "epoch59_acc0.9205.pth" in RAFDB, results are as follow：test accuracy is 0.9192 with test batch-size=64,shuffle=False;

**Comment by pengxj** [comment]
" each batch just contain one class" can obviously raise the acc. number. However, as I said, this is batch recognition since you use a batch of sample features for recognition in test which is not the case in practice.

**Comment by Epicghx** [comment]
Have you obtained the accuracy close to 65.2 when training on AffectNet?
The results I retraining on AffectNet is 5 points lower than the given.


> You are right. with the addition of shuffle, there are fluctuations in accuracy, but they are not always degraded (trained on affectnet).
> And I tried that, each batch just contain one class, then acc will improve to 85.0+.

**Comment by elk-april** [comment]
> Have you obtained the accuracy close to 65.2 when training on AffectNet?
> The results I retraining on AffectNet is 5 points lower than the given.
> 
> > You are right. with the addition of shuffle, there are fluctuations in accuracy, but they are not always degraded (trained on affectnet).
> > And I tried that, each batch just contain one class, then acc will improve to 85.0+.

No, 64.2 is the highest acc I got.

**Comment by uaiaaaa** [comment]
Same question here. I believe that we should not change the test pipeline and use information from other images when evaluating the performance of the trained model.

**Comment by pengxj** [comment]
> Same question here. I believe that we should not change the test pipeline and use information from other images when evaluating the performance of the trained model.

that's it.

**Comment by JiaweiShiCV** [comment]
Thank you for finding the defect. A solution to the problem of Sharing Affinity block will be proposed in the near future. Prior to that, the other two blocks (FA and DA) in the paper can be used, independent of the SA block.

**Comment by nothingeasy** [comment]
any progress ?

**Comment by purpleleaves007** [comment]
The authors did not disrupt the order of data reading during the test, so the samples of a batch are from only one class, and the features of each sample plus the average features of the batch will naturally improve the accuracy.  The rest of the modules used in the paper did not work.

**Comment by purpleleaves007** [comment]
The authors did not disrupt the order of data reading during the test, so the samples of a batch are from only one class, and SA makes the features of each sample plus the average features of the batch can naturally improve the accuracy.  The accuracy improvement comes mainly from the SA module, and the rest of the modules used in the paper did not work. This is why the training process does not see the gain, and the test does not work when the batch size is 1.

**Comment by zengqunzhao** [comment]
> And I think if shuffe=True in testing, the performance will degrade seriously.

Yes, exactly. The test results in the paper are not convincing. If the shuffle=True(it should be True), the performance will decrease dramatically.
